### ISSUE TITLE
Improve experience of actions that require additional scopes

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#3623]
## Feature Description

Certain API requests in Site Kit require additional scopes than those required for the minimum needed for module setup and function which are generally read-only. E.g. creating a new Analytics property.

When the request is made, Site Kit will return a special error if the user has not granted sufficient permissions to perform the action and is prompted with an action-specific modal before they confirm to be sent through the oAuth scope request/grant flow and back before they can retry.

In most cases, we have sufficient information to know that this will happen before it does because Site Kit is aware of the scopes that have been granted since the last page load. With this information, we should be able to set expectations ahead of time about the need to grant additional permissions or even bypass the need for the permissions modal entirely with preemptive messaging. We already do this for Analytics account provisioning as shown below:
![image](https://user-images.githubusercontent.com/1621608/123122085-61e70600-d44e-11eb-84fc-c2a3cba9de61.png)

In this circumstance, we do not show the permission scope modal and instead send the user directly to oAuth.

In either case, we should reevaluate this flow to better inform the user (similar to above) as to when we already know that additional permissions are necessary rather than "surprising" them with it at the time of taking action.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
@ivonac4 it's still relevant, yes. For the most part, this is the approach we have been taking for a while now I feel, so I suppose this issue is about back porting that treatment to flows which have been around longer and still use the older on-demand dialog.

**Comment by aaemnnosttv** [comment]
I've added AC here but this is blocked until we have designs to reference how and where the respective notices should appear and the language to use. Once we have a design to reference, this should be ready for IB.

**Comment by sigal-teller** [comment]
Hey @aaemnnosttv If I understand correctly you are referring to a notice that would be added during certain module setups and will inform the user that the next step would be granting SK permissions?
As for the language we use, I thought this should be handled more holistically, especially since we're working on the setup flow refresh, which also includes setup of these permissions as well.
Fo you prefer we address this separately? I know that this issue refers not only to the initial SK setup but can potentially occur while setting up specific modules. I was referring to using similar language in both use cases.

**Comment by aaemnnosttv** [comment]
> If I understand correctly you are referring to a notice that would be added during certain module setups and will inform the user that the next step would be granting SK permissions?

@sigal-teller Yes, basically informing users before they take the action that would require additional permissions, rather than interrupting them with the modal as we have in some cases. This is how we have been doing it already for new features, so this is about applying that consistently. E.g. things like creating a new GA property, or web data stream, or containers in Tag Manager all use the old flow I believe.
